armgcc.lua

```diff
@@ -21,10 +21,17 @@ function use_toolchain(sdk_path)
 		set_kind("cross")
 		set_description("Stm32 Arm Embedded Compiler")
 		set_sdkdir(sdk_path)
-		set_toolset("cc", "arm-none-eabi-gcc")
-		set_toolset("ld", "arm-none-eabi-gcc")
-		set_toolset("ar", "arm-none-eabi-ar")
-		set_toolset("as", "arm-none-eabi-gcc")
+		if is_plat("windows") then
```

Comment:
you should set toolset and call toolchain:is_plat in on_load.

https://github.com/xmake-io/xmake/blob/2c973f5431c7fc907ed52a01698f1dd51804507d/xmake/toolchains/armcc/xmake.lua#L46

```lua
on_load(function (toolchain)
    if toolchain:is_plat("windows") then
        toolchain:set("toolset", "cc", "...")
    end
end)